lib/babe/build_integration_test.go

```diff
@@ -139,6 +139,49 @@ func TestBuildBlock_ok(t *testing.T) {
 	require.Equal(t, 1, len(extsBytes))
 }
 
+func TestBuildBlock_addToTransactionState(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	telemetryMock := NewMockClient(ctrl)
+	telemetryMock.EXPECT().SendMessage(gomock.Any()).AnyTimes()
+
+	cfg := &ServiceConfig{
+		TransactionState: state.NewTransactionState(telemetryMock),
+		LogLvl:           log.Info,
+	}
+
+	babeService := createTestService(t, cfg)
+
+	parentHash := babeService.blockState.GenesisHash()
+	rt, err := babeService.blockState.GetRuntime(nil)
+	require.NoError(t, err)
+
+	epochData, err := babeService.initiateEpoch(testEpochIndex)
+	require.NoError(t, err)
+
+	ext := runtime.NewTestExtrinsic(t, rt, parentHash, parentHash, 0, "System.remark", []byte{0xab, 0xcd})
+	block := createTestBlock(t, babeService, emptyHeader, [][]byte{common.MustHexToBytes(ext)},
+		1, testEpochIndex, epochData)
```

Comment:
`block` is never used

Also I'm not a genius, my linter is, you might want to bump your linter up 😉 

```
go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.47
```